src/Php80/PhpToken.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Symfony\Polyfill\Php80;
+
+/**
+ * @author Fedonyuk Anton <info@ensostudio.ru>
+ * @internal
+ */
+class PhpToken implements \Stringable
+{
+    /**
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var string
+     */
+    public $text;
+
+    /**
+     * @var int
+     */
+    public $line;
+
+    /**
+     * @var int
+     */
+    public $pos;
+
+    public function __construct(int $id, string $text, int $line = -1, int $position = -1)
+    {
+        $this->id = $id;
+        $this->text = $text;
+        $this->line = $line;
+        $this->pos = $position;
+    }
+
+    public function getTokenName(): ?string
+    {
+        $name = \token_name($this->id);
+        if ($name === 'UNKNOWN') {
+            $name = \strlen($this->text) > 1 || \ord($this->text) < 32 ? null : $this->text;
+        }
+        return $name;
+    }
+
+    /**
+     * @param int|string|array $kind
+     */
+    public function is($kind): bool
+    {
+        $kind = (array) $kind;
+        if (\is_int(\current($kind))) {
```

Comment:
won't this fail when there are more that one item in $kind and not all of them are integers?
can you also add a test case that covers the `getTokenName` check?

Reply:
@nicolas-grekas yes, need rewrite and test every value

Reply:
@nicolas-grekas fixed. In self defense: that such a possibility is undocumented. :)

Reply:
@nicolas-grekas 
 > can you also add a test case that covers the getTokenName check?

removed, [native class don't use it](https://3v4l.org/It16L#v8.1.3)